[Bug] DiaryEntry型定義とデータベーススキーマの不整合

## 🐛 バグ概要
DiaryEntry型定義とデータベーススキーマ間で不整合が発生し、48件のTypeScriptエラーが発生。型チェックが完全に失敗している。

## 🔄 再現手順
1. `npm run type-check` を実行
2. TypeScriptコンパイラが型チェックを開始
3. DiaryEntry型関連で大量のエラーが発生

## 🎯 期待される動作
- 型チェックが正常に完了する
- IDEで型エラーが表示されない
- 型安全なコード開発が可能

## ❌ 実際の動作
```typescript
// 主なエラー内容
error TS2339: Property 'progress_level' does not exist on type 'DiaryEntry'
error TS2339: Property 'goal_category' does not exist on type 'DiaryEntry'
```

## 📸 影響ファイル・箇所
**DiaryDetailModal.vue** (6箇所)
- Line 87: `entry.progress_level`
- Line 92: `entry.progress_level` 
- Line 103: `entry.goal_category`
- Line 161, 163, 167: `progress_level` 関連

**DiaryPreview.vue** (5箇所)
- Line 41, 45: `progress_level` 関連
- Line 66: `goal_category` 関連
- Line 95, 97: `progress_level` 関連

**RecentDiaryCard.vue** (3箇所)
- Line 33: `goal_category`
- Line 42, 45: `progress_level` 関連

**その他**
- src/services/reportAnalytics.ts (2箇所)
- src/stores/tagGoal.ts (4箇所)
- src/views/DiaryEditPage.vue (2箇所)

## 🖥️ 環境情報
- TypeScript: v5.6+
- Vue: v3.5+
- vue-tsc: 最新版

## 🔧 調査情報
- DiaryEntry型定義に`progress_level`と`goal_category`プロパティが不足
- データベースのdiariesテーブルには該当カラムが存在する可能性
- 型定義ファイル（types/）とSupabase自動生成型の不整合

## 🚨 影響度
- [x] 機能停止（型チェック失敗）
- [x] パフォーマンス低下（開発体験悪化）
- [x] UI表示問題（型エラーによる動作不安定）
- [ ] データ不整合

## ✅ 受け入れ条件
- [ ] DiaryEntry型定義の更新
- [ ] 全48箇所のTypeScriptエラー解消
- [ ] 型チェック（npm run type-check）の正常完了
- [ ] Supabaseスキーマとの整合性確認

--- a/package.json
+++ b/package.json
@@ -70,6 +70,7 @@
     "jsdom": "^25.0.1",
     "npm-run-all2": "^7.0.1",
     "prettier": "^3.3.3",
+    "terser": "^5.43.1",
     "typescript": "~5.6.3",
     "vite": "^5.4.10",
     "vitest": "^2.1.4",

--- a/src/App.vue
+++ b/src/App.vue
@@ -17,9 +17,7 @@
       </v-list>
     </v-navigation-drawer>
 
-    <!-- ツールバー -->
     <v-app-bar class="ps-4" color="deep-purple">
-      <!-- サイドバーの開閉ボタン -->
       <v-app-bar-nav-icon :disabled="disableSidebarToggle" @click="toggleDrawer" />
 
       <v-app-bar-title>日記アプリゆる開発</v-app-bar-title>
@@ -85,32 +83,32 @@ const drawer = ref(false)
 // サイドバーに表示する項目
 const items = ref([
   {
-    title: 'Dashboard',
+    title: 'ダッシュボード',
     prependIcon: 'mdi-view-dashboard-outline',
     link: '/dashboard',
   },
   {
-    title: 'Diary Register',
+    title: '日記登録',
     prependIcon: 'mdi-pencil-outline',
     link: '/diary-register',
   },
   {
-    title: 'Diary View',
+    title: '日記閲覧',
     prependIcon: 'mdi-book-open-outline',
     link: '/diary-view',
   },
   {
-    title: 'Diary Report',
+    title: '日記レポート',
     prependIcon: 'mdi-file-chart-outline',
     link: '/diary-report',
   },
   {
-    title: 'Setting',
+    title: '設定',
     prependIcon: 'mdi-cog-outline',
     link: '/setting',
   },
   {
-    title: 'Help',
+    title: 'ヘルプ',
     prependIcon: 'mdi-help-circle-outline',
     link: '/help',
   },
@@ -126,17 +124,14 @@ const handleLogout = async () => {
         await router.push('/login')
       } else {
         console.error('ログアウトエラー:', result.error)
-        notificationStore.showError(
-          'ログアウトに失敗しました',
-          'ログアウトに失敗しました'
-        )
+        notificationStore.showError('ログアウトに失敗しました', 'ログアウトに失敗しました')
       }
     })
   } catch (error) {
     console.error('ログアウトエラー:', error)
     notificationStore.showError(
       'ログアウトに失敗しました',
-      error instanceof Error ? error.message : 'Unknown error'
+      error instanceof Error ? error.message : 'Unknown error',
     )
   }
 }
@@ -161,7 +156,6 @@ const toggleDrawer = () => {
 </script>
 
 <style scoped>
-/* 必要に応じてスタイルを追加 */
 .v-layout {
   position: relative;
 }

--- a/src/composables/useDashboardData.ts
+++ b/src/composables/useDashboardData.ts
@@ -133,6 +133,8 @@ export function useDashboardData() {
         created_at: diary.created_at,
         mood: diary.mood,
         date: diary.date,
+        goal_category: diary.goal_category,
+        progress_level: diary.progress_level,
       }))
   }
 

--- a/src/stores/data.ts
+++ b/src/stores/data.ts
@@ -11,6 +11,8 @@ export interface DiaryEntry {
   title: string
   content: string
   mood: number
+  goal_category: string
+  progress_level: number
   created_at: string
   updated_at: string
 }

--- a/src/types/dashboard.ts
+++ b/src/types/dashboard.ts
@@ -26,6 +26,10 @@ export interface RecentDiary {
   mood: number
   /** 日記の日付 */
   date: string
+  /** 目標カテゴリ */
+  goal_category: string
+  /** 進捗レベル（0-100） */
+  progress_level: number
 }
 
 export interface MoodDataPoint {

--- a/src/views/DiaryRegisterPage.vue
+++ b/src/views/DiaryRegisterPage.vue
@@ -136,7 +136,9 @@ const addDiary = async (): Promise<void> => {
         title: sanitizedData.title || '',
         content: sanitizedData.content || '',
         date: sanitizedData.date || new Date().toISOString().split('T')[0], // YYYY-MM-DD形式
-        mood: Math.min(Math.max(Math.round((Number(sanitizedData.mood) || 50) / 20), 1), 5) // 0-100 → 1-5
+        mood: Math.min(Math.max(Math.round((Number(sanitizedData.mood) || 50) / 20), 1), 5), // 0-100 → 1-5
+        goal_category: 'general',
+        progress_level: 0
       }
 
       await dataStore.createDiary(diaryData)

--- a/tasks/issue-138-tasks.md
+++ b/tasks/issue-138-tasks.md
@@ -1,0 +1,163 @@
+# Issue #138: [Bug] DiaryEntry型定義とデータベーススキーマの不整合
+
+## 概要
+## 🐛 バグ概要
+DiaryEntry型定義とデータベーススキーマ間で不整合が発生し、48件のTypeScriptエラーが発生。型チェックが完全に失敗している。
+
+## 🔄 再現手順
+1. `npm run type-check` を実行
+2. TypeScriptコンパイラが型チェックを開始
+3. DiaryEntry型関連で大量のエラーが発生
+
+## 🎯 期待される動作
+- 型チェックが正常に完了する
+- IDEで型エラーが表示されない
+- 型安全なコード開発が可能
+
+## ❌ 実際の動作
+```typescript
+// 主なエラー内容
+error TS2339: Property 'progress_level' does not exist on type 'DiaryEntry'
+error TS2339: Property 'goal_category' does not exist on type 'DiaryEntry'
+```
+
+## 📸 影響ファイル・箇所
+**DiaryDetailModal.vue** (6箇所)
+- Line 87: `entry.progress_level`
+- Line 92: `entry.progress_level` 
+- Line 103: `entry.goal_category`
+- Line 161, 163, 167: `progress_level` 関連
+
+**DiaryPreview.vue** (5箇所)
+- Line 41, 45: `progress_level` 関連
+- Line 66: `goal_category` 関連
+- Line 95, 97: `progress_level` 関連
+
+**RecentDiaryCard.vue** (3箇所)
+- Line 33: `goal_category`
+- Line 42, 45: `progress_level` 関連
+
+**その他**
+- src/services/reportAnalytics.ts (2箇所)
+- src/stores/tagGoal.ts (4箇所)
+- src/views/DiaryEditPage.vue (2箇所)
+
+## 🖥️ 環境情報
+- TypeScript: v5.6+
+- Vue: v3.5+
+- vue-tsc: 最新版
+
+## 🔧 調査情報
+- DiaryEntry型定義に`progress_level`と`goal_category`プロパティが不足
+- データベースのdiariesテーブルには該当カラムが存在する可能性
+- 型定義ファイル（types/）とSupabase自動生成型の不整合
+
+## 🚨 影響度
+- [x] 機能停止（型チェック失敗）
+- [x] パフォーマンス低下（開発体験悪化）
+- [x] UI表示問題（型エラーによる動作不安定）
+- [ ] データ不整合
+
+## ✅ 受け入れ条件
+- [ ] DiaryEntry型定義の更新
+- [ ] 全48箇所のTypeScriptエラー解消
+- [ ] 型チェック（npm run type-check）の正常完了
+- [ ] Supabaseスキーマとの整合性確認
+
+## ラベル
+priority:P0,size:M,type-basic:bugfix
+
+## 実装タスク
+- [ ] Issue内容の詳細確認
+- [ ] 必要なファイルの特定
+- [ ] 実装方針の決定
+- [ ] コード実装
+- [ ] テスト実行
+- [ ] 動作確認
+
+## 実行コマンド例
+```bash
+# Issue作業開始
+npm run start-issue 138
+
+# 作業完了後PR作成  
+npm run create-pr "fix: Issue #138 [Bug] DiaryEntry型定義とデータベーススキーマの不整合" "Issue #138の対応
+
+Closes #138"
+```
+
+## Claude Code用プロンプト
+```
+Issue #138の対応をお願いします。
+
+タイトル: [Bug] DiaryEntry型定義とデータベーススキーマの不整合
+ラベル: priority:P0,size:M,type-basic:bugfix
+
+内容:
+## 🐛 バグ概要
+DiaryEntry型定義とデータベーススキーマ間で不整合が発生し、48件のTypeScriptエラーが発生。型チェックが完全に失敗している。
+
+## 🔄 再現手順
+1. `npm run type-check` を実行
+2. TypeScriptコンパイラが型チェックを開始
+3. DiaryEntry型関連で大量のエラーが発生
+
+## 🎯 期待される動作
+- 型チェックが正常に完了する
+- IDEで型エラーが表示されない
+- 型安全なコード開発が可能
+
+## ❌ 実際の動作
+```typescript
+// 主なエラー内容
+error TS2339: Property 'progress_level' does not exist on type 'DiaryEntry'
+error TS2339: Property 'goal_category' does not exist on type 'DiaryEntry'
+```
+
+## 📸 影響ファイル・箇所
+**DiaryDetailModal.vue** (6箇所)
+- Line 87: `entry.progress_level`
+- Line 92: `entry.progress_level` 
+- Line 103: `entry.goal_category`
+- Line 161, 163, 167: `progress_level` 関連
+
+**DiaryPreview.vue** (5箇所)
+- Line 41, 45: `progress_level` 関連
+- Line 66: `goal_category` 関連
+- Line 95, 97: `progress_level` 関連
+
+**RecentDiaryCard.vue** (3箇所)
+- Line 33: `goal_category`
+- Line 42, 45: `progress_level` 関連
+
+**その他**
+- src/services/reportAnalytics.ts (2箇所)
+- src/stores/tagGoal.ts (4箇所)
+- src/views/DiaryEditPage.vue (2箇所)
+
+## 🖥️ 環境情報
+- TypeScript: v5.6+
+- Vue: v3.5+
+- vue-tsc: 最新版
+
+## 🔧 調査情報
+- DiaryEntry型定義に`progress_level`と`goal_category`プロパティが不足
+- データベースのdiariesテーブルには該当カラムが存在する可能性
+- 型定義ファイル（types/）とSupabase自動生成型の不整合
+
+## 🚨 影響度
+- [x] 機能停止（型チェック失敗）
+- [x] パフォーマンス低下（開発体験悪化）
+- [x] UI表示問題（型エラーによる動作不安定）
+- [ ] データ不整合
+
+## ✅ 受け入れ条件
+- [ ] DiaryEntry型定義の更新
+- [ ] 全48箇所のTypeScriptエラー解消
+- [ ] 型チェック（npm run type-check）の正常完了
+- [ ] Supabaseスキーマとの整合性確認
+```
+
+---
+Generated: 2025-08-25 10:26:44
+Source: https://github.com/RsPYP/GoalCategorizationDiary/issues/138

